libs/obs/src/CActionRobotMovement3D.cpp

```diff
@@ -93,3 +95,111 @@ void  CActionRobotMovement3D::readFromStream(mrpt::utils::CStream &in, int versi
 
 }
 
+
+void	CActionRobotMovement3D::computeFromOdometry(
+	const CPose3D				&odometryIncrement,
+	const TMotionModelOptions	&options)
+{
+	// Set the members data:
+	estimationMethod			= emOdometry;
+	rawOdometryIncrementReading	= odometryIncrement;
+	motionModelConfiguration	= options;
+
+	if ( options.modelSelection == mm6DOF )
+				computeFromOdometry_model6DOF( odometryIncrement, options );
+}
+
+
+/*---------------------------------------------------------------
+						TMotionModelOptions
+ ---------------------------------------------------------------*/
+CActionRobotMovement3D::TMotionModelOptions::TMotionModelOptions() :
+modelSelection(mm6DOF),
+	mm6DOFModel()
+{
+
+	mm6DOFModel.nParticlesCount		= 300;
+	mm6DOFModel.a1		= 0.05f;
+	mm6DOFModel.a2		= 0.05f;
+	mm6DOFModel.a3		= 0.05f;
+	mm6DOFModel.a4		= 0.05f;
+	mm6DOFModel.a5		= 0.05f;
+	mm6DOFModel.a6		= 0.05f;
+	mm6DOFModel.a7		= 0.05f;
+	mm6DOFModel.a8		= 0.05f;
+	mm6DOFModel.a9		= 0.05f;
+	mm6DOFModel.a10		= 0.05f;
+	mm6DOFModel.additional_std_XYZ	= 0.001f;
+	mm6DOFModel.additional_std_angle	= DEG2RAD(0.05f);
+
+}
+
+/*---------------------------------------------------------------
+				computeFromOdometry_model6DOF
+	---------------------------------------------------------------*/
+void	CActionRobotMovement3D::computeFromOdometry_model6DOF(
+	const CPose3D				&odometryIncrement,
+	const TMotionModelOptions	&o
+	)
+{
+
+
+	// The Gaussian PDF:
+	// ---------------------------
+	CPose3DPDFParticles		*aux;
+	static CPose3D			nullPose(0,0,0,0,0,0);
+
+	poseChangeTemp = CPose3DPDFParticles::Create();
+	aux = static_cast<CPose3DPDFParticles*>( poseChangeTemp.pointer() );
+
+	// Set the number of particles:
+	aux->resetDeterministic(nullPose, o.mm6DOFModel.nParticlesCount );
+
+
+
+	//The motion model: A.L. Ballardini [and colleagues]. An effective 6DoF motion model for 3D-6DoF Monte Carlo Localization
```

Comment:
Yes! And add a very brief text explanation of when and what is that model used for (in what method call,...), complete the citation with the full paper reference, i.e:

A. L. Ballardini, A. Furlan, A. Galbiati, M. Matteucci, F. Sacchi, D. G. Sorrenti An effective 6DoF motion model for 3D-6DoF Monte Carlo Localization 4th Workshop on Planning, Perception and Navigation for Intelligent Vehicles, IROS, 2012

Thanks!